doc/src/guide.rst

```diff
@@ -279,52 +284,86 @@ Functions
 
 How to create a new function with one variable::
 
-    class sign(Function):
-
-        nargs = 1
+    >>> from sympy import Mul, S
+    >>> class sign(Function):
+    ...     nargs = 1
+    ...
+    ...     @classmethod
+    ...     def eval(cls, arg):
+    ...         if arg is S.NaN:
+    ...             return S.NaN
+    ...         if arg is S.Zero:
+    ...             return S.Zero
+    ...         if arg.is_positive:
+    ...             return S.One
+    ...         if arg.is_negative:
+    ...             return S.NegativeOne
+    ...         if isinstance(arg, Mul):
+    ...             coeff, terms = arg.as_coeff_mul()
+    ...             if coeff is not S.One:
+    ...                 return cls(coeff) * cls(Mul(*terms))
+    ...
+    ...     def _eval_is_finite(self):
+    ...         return True
+    ...
+    ...     def _eval_conjugate(self):
+    ...         return self
 
-        @classmethod
-        def eval(cls, arg):
-            if arg is S.NaN:
-                return S.NaN
-            if arg is S.Zero:
-                return S.Zero
-            if arg.is_positive:
-                return S.One
-            if arg.is_negative:
-                return S.NegativeOne
-            if isinstance(arg, Basic.Mul):
-                coeff, terms = arg.as_coeff_mul()
-                if not isinstance(coeff, Basic.One):
-                    return cls(coeff) * cls(Basic.Mul(*terms))
+The applied function ``sign(x)`` is constructed using
 
-        is_finite = True
+    >>> sign(x)
+    sign(x)
 
-        def _eval_conjugate(self):
-            return self
+both inside and outside of SymPy. Unapplied functions ``sign`` is just the class
+itself::
 
-        def _eval_is_zero(self):
-            return isinstance(self[0], Basic.Zero)
+    >>> sign
+    sign
 
-and that's it. The ``_eval_*`` functions are called when something is needed.
 The ``eval`` is called when the class is about to be instantiated and it
-should return either some simplified instance of some other class or if the
-class should be unmodified, return ``None`` (see ``core/function.py`` in
-``Function.__new__`` for implementation details). See also tests in
-`sympy/functions/elementary/tests/test_interface.py
-<https://github.com/sympy/sympy/blob/master/sympy/functions/elementary/tests/test_interface.py>`_ that test this interface. You can use them to create your own new functions.
+should return either an another SymPy object or ``None``.
+If it is returned ``None``, it becomes a
+(see ``core/function.py`` in ``Function.__new__`` for implementation details)
 
-The applied function ``sign(x)`` is constructed using
-::
+Here are some examples that how ``eval`` works
+
```

Comment:
I think `::` is added to create a block in `rst` document. Should be done to separate the code block here.